src/backend/main.ts

```diff
@@ -160,6 +165,76 @@ export async function bootstrapWorkspace(opts: {
     }
     yDoc.clientID = clientID
   }
+  // Adam Chen - Injection point 4: new member join
+  console.log('-- Injection point 4: new member join check --')
+  const appPrefixName = appPrefix.toString()
+  console.log('debug: appPrefix: ', appPrefixName)
+  console.log('debug: nodeID', nodeId)
+  const snapshotName = appPrefix.append('32=snapshot').toString()
+  console.log('debug: theoretical snapshot prefix: ', snapshotName)
+  const localYJSUpdate = await persistStore.get('localSnapshot')
+  console.log('persistent storage local yjs update: ', localYJSUpdate)
+  // localYJSUpdate = undefined
+
+  // //State Vector Testing
+  // let test = new StateVector()
+  // test.set(new Name('/user1'), 1)
+  // test.set(new Name('/user2'), 1)
+  // let svsComponent = new Component(Version.type, Encoder.encode(test))
+  // console.log('Debug: SVS test: ', svsComponent.toString())
+  // console.log('Debug: parsing SVS', Decoder.decode(svsComponent.value, StateVector))
+
+  if (!localYJSUpdate) {
+    console.log('Empty yjs local storage detected, fetching snapshot')
+    const interest = new Interest(snapshotName, Interest.CanBePrefix, Interest.MustBeFresh)
+    try {
+      const data = await consume(interest)
+      console.log(`Received data with name [${data.name}]`)
+      const targetName = data.name.getPrefix(-1)
+      // /grpPrefix/32=snapshot/54=<vector>/
+      console.log('segmented object fetcher targeting name: ', targetName.toString())
+      // code from SyncAgent
+      const buffers = []
+      try {
+        console.log('SegmentedObject fetching a snapshot at part 4')
+        const result = fetch(targetName, {
+          verifier: certStorage.verifier, // we have access to the verifier
+          modifyInterest: { mustBeFresh: true },
+          lifetimeAfterRto: 2000,
+          retxLimit: 150, // See Deliveries. 60*1000/(2*200)=150. Default minRto = 150.
+        })
+        for await (const segment of result) {
+          // Cache packets temporary disabled.
+          // this.persistStorage.set(segment.name.toString(), Encoder.encode(segment));
+          // Reassemble
+          buffers.push(segment.content)
+        }
+      } catch (e: any) {
+        console.error(`Unable to fetch ${targetName}: `, e)
```

Comment:
Please stop and move to normal workspace flow. You cannot continue to use incomplete `buffers`.
Also, I believe `fetch` could return a concatenated result for you, so you don't have to concat them yourself.

Reply:
This is a copy paste of how a blob is fetched by syncAgent. 
But it is correct, I should have it exit all the way.
Will move the fetch outside and have a look at the fetch code.

Reply:
No need to apology :). Just remember to fix before you mark the draft as ready.